src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2011,7 +2023,7 @@ private void AddAssemblyAttributesToTable()
             }
 
             AddAssemblyAttributesToTable(
-                this.module.GetSourceAssemblyAttributes(),
+                this.module.GetSourceAssemblyAttributes(isRefAssembly: Context.ExcludePrivateMembers),
```

Comment:
>Context.ExcludePrivateMembers [](start = 71, length = 29)

Should we check MetadataOnly as well? Consider at least asserting that it is true when ExcludePrivateMembers is true.